### PR TITLE
[NativeAOT] Make Frozen collections tests pass with NativeAOT

### LINKED ISSUE #78046: Frozen collection tests fail on NativeAOT
Failures in https://github.com/dotnet/runtime/pull/77934

Either the tests should be made compatible with NativeAOT (preferrable, if possible), or disabled in the NativeAOT test runs.

The typical failure looks like:

```
[FAIL] System.Collections.Frozen.Tests.FrozenDictionary_Generic_Tests_string_string_Ordinal.LookupItems_AllItemsFoundAsExpected(size: 1, comparer: null, specifySameComparer: False)
System.NotSupportedException : 'Xunit.Sdk.AssertEqualityComparer`1[System.Collections.Generic.HashSet`1[System.Collections.Generic.KeyValuePair`2[System.String, System.String]]].CompareTypedSets[System.Collections.Generic.KeyValuePair`2[System.String, System.String]](System.Collections.IEnumerable,System.Collections.IEnumerable)' is missing native code. MethodInfo.MakeGenericMethod() is not compatible with AOT compilation. Inspect and fix AOT related warnings that were generated when the app was published. For more information see https://aka.ms/nativeaot-compatibility
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.GetUncachedMethodInvoker(RuntimeTypeInfo[], MemberInfo) + 0x30
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.MakeGenericMethod(Type[]) + 0x1c0
   at System.Collections.Frozen.Tests.FrozenDictionary_Generic_Tests`2.LookupItems_AllItemsFoundAsExpected(Int32 size, IEqualityComparer`1 comparer, Boolean specifySameComparer) + 0x280
   at System.Collections.Immutable!<BaseAddress>+0x83e170
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
```

Something is using `MethodInfo.MakeGenericMethod()` and it is not an AOT-friendly pattern.

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by VSadov**
Fixes: #78046

### CODE REVIEW

**Comment by VSadov** [comment]
The OSX arm64 failure is https://github.com/dotnet/runtime/issues/76501

**Comment by VSadov** [comment]
@MichalStrehovsky - can you take a look? This is just the rd.xml change. Thanks!

File: src/libraries/System.Collections.Immutable/tests/default.rd.xml [hunk 1]
```diff
@@ -23,5 +23,46 @@
...
+      </Type>
+    </Assembly>
+
+    <Assembly Name="xunit.assert">
+      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.UInt64,System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
+        <Method Name="CompareTypedSets" Dynamic="Required All">
+          <GenericArgument Name="System.UInt64,System.Private.CoreLib" />
+        </Method>
+      </Type>
+    </Assembly>
```
Comment (MichalStrehovsky) [inline src/libraries/System.Collections.Immutable/tests/default.rd.xml:65]: Nit: we can avoid opening and closing the `Assembly` tag to make things more compact.

```suggestion
    <Assembly Name="xunit.assert">
      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.Collections.Generic.KeyValuePair`2[[System.String, System.Private.CoreLib],[System.String, System.Private.CoreLib]],System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
        <Method Name="CompareTypedSets" Dynamic="Required All">
          <GenericArgument Name="System.Collections.Generic.KeyValuePair`2[[System.String,System.Private.CoreLib],[System.String,System.Private.CoreLib]],System.Private.CoreLib" />
        </Method>
      </Type>

      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.Collections.Generic.KeyValuePair`2[[System.UInt64, System.Private.CoreLib],[System.UInt64, System.Private.CoreLib]],System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
        <Method Name="CompareTypedSets" Dynamic="Required All">
          <GenericArgument Name="System.Collections.Generic.KeyValuePair`2[[System.UInt64,System.Private.CoreLib],[System.UInt64,System.Private.CoreLib]],System.Private.CoreLib" />
        </Method>
      </Type>

      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.Collections.Generic.KeyValuePair`2[[System.Int32, System.Private.CoreLib],[System.Int32, System.Private.CoreLib]],System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
        <Method Name="CompareTypedSets" Dynamic="Required All">
          <GenericArgument Name="System.Collections.Generic.KeyValuePair`2[[System.Int32,System.Private.CoreLib],[System.Int32,System.Private.CoreLib]],System.Private.CoreLib" />
        </Method>
      </Type>

      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.Int32,System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
        <Method Name="CompareTypedSets" Dynamic="Required All">
          <GenericArgument Name="System.Int32,System.Private.CoreLib" />
        </Method>
      </Type>

      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.UInt64,System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
        <Method Name="CompareTypedSets" Dynamic="Required All">
          <GenericArgument Name="System.UInt64,System.Private.CoreLib" />
        </Method>
      </Type>
    </Assembly>
```

**Review by MichalStrehovsky — APPROVED** [review]
Looks good!

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**